manifests/config/context/resource.pp

```diff
@@ -0,0 +1,68 @@
+# Definition: tomcat::config::server::connector
+#
+# Configure Connector elements in $CATALINA_BASE/conf/server.xml
+#
+# Parameters:
+# - $catalina_base is the base directory for the Tomcat installation.
+# - $connector_ensure specifies whether you are trying to add or remove the
+#   Connector element. Valid values are 'true', 'false', 'present', and
+#   'absent'. Defaults to 'present'.
+# - The $port attribute. This attribute is required unless $connector_ensure
+#   is set to false.
+# - The $protocol attribute. Defaults to $name when not specified.
+# - $parent_service is the Service element this Connector should be nested
+#   beneath. Defaults to 'Catalina'.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
+define tomcat::config::context::resource (
+  $resource_name,
+  $auth,
+  $type,
+  $driverClassName,
+  $username,
+  $password,
+  $maxActive,
+  $maxIdle,
+  $maxWait,
+  $url,
+  $factory,
```

Comment:
It doesn't look like all these parameters are required. Anything that isn't actually required for the configuration of the Resource elements should be optional.